docs/design/ftlib.md

```diff
@@ -19,32 +19,43 @@ Additionally, since most communication libraries require rank assigned to each w
 
 *Fig 2. The Role FTLib Played in Distributed Training*
 
-In the contemporary design, the FTLib is called explicitly to perform collective operation on all parameters of a model.
+In the contemporary design, the FTLib is called explicitly to perform collective operations or self-designed high-level calls.
 
-For each batch of data, the model in each worker will be fed with training data, producing loss by forwarding and gradients by backward propagation. With local gradients ready on each worker, the allreduce API in FTLib will be called to average the gradients across all workers. As the operation is blocking, it indicates all parameters on this process is ready to be updated if allreduce API returns success. Otherwise, the training process will abort or continue according to different returned status.
+For each batch of data, the model in each worker will be fed with training data, producing loss by forwarding and gradients by backward propagation. With local gradients ready on each worker, the allreduce API in FTLib will be called to average the gradients across all workers. As the operation is blocking, it indicates all parameters on this process is ready to be updated if allreduce API returns success.
 
-Before launching any communication operations, the FTLib will lock the process and unlock after finished or aborted. This design prevents the initialized flag from being modified by `Consensus.new_member_join` on other thread during the average phase.
+Otherwise, the training process will abort or continue according to different returned status.
 
-The initialized flag of a FTLib will be reset (to `False`) under three occasions:
+The entire step described above can be wrapped into a high-level customized API with FTLib, which hides the latency of communication by overlapping computation and communication.
```

Comment:
I think this is what you meant by "self-designed high-level calls" that you mentioned earlier. Maybe explain this benefit earlier so readers would have more context